contracts/NonfungiblePositionManager.sol

```diff
@@ -187,22 +151,14 @@ contract NonfungiblePositionManager is INonfungiblePositionManager, BaseLiquidit
     }
 
     // TODO: in v3, we can partially collect fees, but what was the usecase here?
-    function collect(uint256 tokenId, address recipient, bytes calldata hookData, bool claims)
-        public
-        returns (BalanceDelta delta)
-    {
+    function collect(uint256 tokenId, address recipient, bytes calldata hookData, bool claims) external {
         TokenPosition memory tokenPos = tokenPositions[tokenId];
-        if (manager.isUnlocked()) {
-            _collect(recipient, tokenPos.range, hookData);
-        } else {
-            bytes[] memory data = new bytes[](1);
-            data[0] = abi.encodeWithSelector(this.collect.selector, tokenId, recipient, hookData, claims);
-
-            Currency[] memory currencies = new Currency[](2);
-            currencies[0] = tokenPos.range.poolKey.currency0;
-            currencies[1] = tokenPos.range.poolKey.currency1;
-            bytes memory result = unlockAndExecute(data, currencies);
-            delta = abi.decode(result, (BalanceDelta));
+
+        // TODO: security -- what happens if PoolManager is already unlocked by UR???
+        _collect(recipient, tokenPos.range, hookData);
+        if (recipient != _msgSenderInternal()) {
```

Comment:
this check be moved inside _collect() imo? any reason why its here?

Reply:
agreed -- should be consistent for the vanilla and batch cases